core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -535,9 +535,10 @@ private[spark] class BlockManager(
               /* We'll store the bytes in memory if the block's storage level includes
                * "memory serialized", or if it should be cached as objects in memory
                * but we only requested its serialized bytes. */
-              val copyForMemory = ByteBuffer.allocate(bytes.limit)
-              copyForMemory.put(bytes)
-              memoryStore.putBytes(blockId, copyForMemory, level)
+              memoryStore.putBytes(blockId, bytes.limit) {
```

Comment:
could you add a // comment here that links to this JIRA? Otherwise it's not clear why we need to do this lazily.